src/postgres/inventory/collection/PgCollectionKey.ts

```diff
@@ -134,10 +135,16 @@ class PgCollectionKey implements CollectionKey<PgClassType.Value, PgCollectionKe
    * @private
    */
   @memoizeMethod
-  private _getSelectLoader (client: Client): DataLoader<PgClassType.Value, PgClassType.Value | null> {
+  private _getSelectLoader (client: Client, resolveInfo: mixed, collectionGqlType: mixed): DataLoader<PgClassType.Value, PgClassType.Value | null> {
     return new DataLoader<PgClassType.Value, PgClassType.Value | null>(
-      async (keys: Array<PgClassType.Value>): Promise<Array<PgClassType.Value | null>> => {
+      async (keysAndStuff: mixed): Promise<Array<PgClassType.Value | null>> => {
```

Comment:
It gets more challenging when you factor in the relation fetching from #342, because then merging the fields from across different places would also merge the subqueries - fetching WAY more data than is necessary.

Since doing more work on #342 I actually think the better way to deal with this is to eliminate DataLoader entirely, and instead to inline all relationships via Join-Monster style queries that just pull down everything for a root node. Then the number of queries would be basically the same as the number of root nodes on the GraphQL query, which I think makes a lot of sense.

I don't think DataLoader is a good match once you start selecting specific fields and adding subqueries or JOINs into the mix